server/src/main/java/com/cloud/network/NetworkServiceImpl.java

```diff
@@ -4264,23 +4265,37 @@ public boolean deletePhysicalNetwork(final Long physicalNetworkId) {
         return Transaction.execute(new TransactionCallback<Boolean>() {
             @Override
             public Boolean doInTransaction(TransactionStatus status) {
-                // delete vlans for this zone
-                List<VlanVO> vlans = _vlanDao.listVlansByPhysicalNetworkId(physicalNetworkId);
-                for (VlanVO vlan : vlans) {
-                    _vlanDao.remove(vlan.getId());
-                }
-
-                // Delete networks
-                List<NetworkVO> networks = _networksDao.listByPhysicalNetwork(physicalNetworkId);
-                if (networks != null && !networks.isEmpty()) {
-                    for (NetworkVO network : networks) {
-                        _networksDao.remove(network.getId());
-                    }
-                }
+                disablePhysicalNetwork(physicalNetworkId, pNetwork);
```

Comment:
Isn't better to reverse the order of these function calls? yes, these are in a transaction but better to be in the reverse flow.

Reply:
please explain how you would like to see the order be, @harikrishna-patnala . 
I created this order as it was and only added the disabling in front. I think the deltion of the ip addresses happens at the right place.

Reply:
Agree with disabling in front @DaanHoogland. may be we can reorder network and ipaddress!